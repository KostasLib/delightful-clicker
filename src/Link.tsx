import React, { FC } from 'react';

interface IProps {
  source: string;
  title?: string;
}

export const Link: FC<IProps> = props => (
  <a target="blank" href={props.source} title={props.title}>
    {props.children}
  </a>
);
