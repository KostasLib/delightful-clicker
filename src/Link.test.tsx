import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Link } from './Link';

describe('The Link component', () => {
  it('renders an Anchor element', () => {
    const wrapper = shallow(
      <Link title="My link test" source="www.github.com/kostaslib">
        My profile
      </Link>,
    );
    const tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
