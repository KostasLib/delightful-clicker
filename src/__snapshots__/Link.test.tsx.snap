// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Link component renders an Anchor element 1`] = `
<a
  href="www.github.com/kostaslib"
  target="blank"
  title="My link test"
>
  My profile
</a>
`;
